java/foam/core2/FObject.java

```diff
@@ -28,6 +28,21 @@
 import java.util.List;
 
 public abstract class FObject implements Cloneable, Comparable, java.io.Serializable {
+
+  private int UID_ = -1;
+  private static int nextUid_ = 1;
```

Comment:
Also make a nextUID_() static method, which returns the nextUid_++, and make that method synchronized to prevent race conditions causing objects to have the same UID.